### ISSUE TITLE
Split DNS

### ISSUE THREAD

**Issue by giox069** [Tunnelblick/Tunnelblick#345]
This is a feature request.
In many enterprise class VPN clients, Split DNS is an option which allows a client to resolve only some DNS domain zones with the DNS servers provided by the OpenVPN server. The rest of DNS queries will be routed to the default client's DNS resolver.
An example: I'm at home, and I have my home DNS, which handle the zone "myhome.lan". So I can resolve the following hosts:
```
  owncloud.myhome.lan   192.168.10.12
  mailserver.myworkdomain.com   88.88.88.88   (a public ip)
```
Now I connect to my OpenVPN server at work. OpenVPN server pushes me "myworkdomain.com" as DOMAIN and a DNS for it.
Tunnelblick will use the new DNS server to resolve both names:
```
owncloud.myhome.lan   77.77.77.77     (my home's public IP address, wrong, because I'm at home!)
mailserver.myworkdomain.com   192.168.70.10    (correct, I can reach enterprise mailserver via VPN tunnel)
mypc.myworkdomain.com    192.168.70.20 (correct, I can reach my PC via VPN)
```
With Split DNS correctly activated, only queries for myworkdomain.com will be routed to the openvpn provided DNSes, so it will correctly work
```
owncloud.myhome.lan  192.168.10.12
mailserver.myworkdomain.com   192.168.70.10
mypc.myworkdomain.com    192.168.70.20
```
Currently there is a low-price commercial openvpn client which does it in an excellent way (viscosity).
But I would like it becomes a standard available on all OpenVPN clients.

**Comment by jkbullard** [comment]
Yes, it would be nice to have split tunnels. It is the subject of numerous discussions on the [Tunnelblick Discussion Group](https://groups.google.com/forum/#!forum/tunnelblick-discuss).

Here are some comments and questions, for anyone interested in helping:
 * It would be implemented in "client.tunnelblick.up.sh" and "client.tunnelblick.down.sh". "up" would set it up, and (if it had been set up by "up")  "down" would un-set it.
 * Options from OpenVPN are made available to the scripts in the "$foreign_option_n" environment variables (foreign_option_1, foreign_option_2, etc.).
 * How can the split tunnel be specified in the OpenVPN configuration file and "pushed" options, and how is that transformed by OpenVPN into "foreign_option_n" variables? Is ordering preserved?
 * Should the configuration file be able to override "pushed" options, or vice versa? If so, how? (That's really an OpenVPN question, I think.)
 * How should a sequence of multiple DOMAIN and DNS options be interpreted? Does that break any existing setups?
 * How can the split dns be implemented on OS X? There is some discussion of this in the Discussion Group.

**Comment by giox069** [comment]
Ok, please forgive me. I already commented for the same request on issue #305 some month ago and then I forgot it. You can close this issue or leave it open as feature request :)
Unfortunately I cannot help you with your 6 questions.
Currently I'm playing with the proprietary VPN client I mentioned on the 1st post, and it "just works".
It has a parameter called "Split DNS mode", which by default is AUTO, but can be changed both locally or pushed by the server. If you look at their site, you will find some useful informations. Google for configuring-dns-and-wins-settings and sparklabs and see the section "DNS Modes". There is not too much, but in my case is enough.
I would not like to cheat/steal their job (I already bought them 4 licenses). I'm just hoping that a coehrent Split DNS will be standardized across many OpenVPN clients.
Just for reference: Ubunu also offers some kind of Split DNS magic with dnsmasq + networkmanager when a OpenVPN or Cisco IPsec VPN (vpnc) are connected. But I cannot find further documentation.

**Comment by jkroepke** [comment]
I would like to see this in Tunnelblick, too!

A 3rd party integration between openvpn and systemd implements split dns using a custom property named `DOMAIN-ROUTE`

https://github.com/jonathanio/update-systemd-resolved

By define `options DOMAIN-ROUTE example.com`, the dns server of the VPN is responsible only for example.com.

This 3rd party integration is installed and enabled by default in distros like [Ubuntu](https://packages.ubuntu.com/focal/openvpn)

**Comment by jkroepke** [comment]
It seems like OpenVPN 2.6 will introduce a better interface for DNS to archive Split DNS.

https://gitlab.com/openvpn/openvpn/-/commit/b3e0d95dcfd0de2a5fe6545fed8f46e0dd35784d

I hope, Tunnelblick will integrate it well.

**Comment by denniseffing** [comment]
Is this still relevant?

`dhcp-option DOMAIN-ROUTE <domain>` seems to work fine.
Not sure regarding the dns options introduced with OpenVPN 2.6.

**Comment by worldowner** [comment]
Yes, it is still relevant. When ```dhcp-option DOMAIN-ROUTE <domain>``` (which is just a name used in [update-systemd-resolved](https://github.com/jonathanio/update-systemd-resolved), it's not standardized) is sent to Tunnelblick it should set DNS pushed by OpenVPN server to be used only for this particular domain (or domains in case when multiple DOMAIN-ROUTEs are pushed). This is not how tunnelblick currently works.

**Comment by denniseffing** [comment]
I see, thanks for clarifying! I can confirm that this is not how Tunnelblick currently works.
